spark/v3.2/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestAddFilesProcedure.java

```diff
@@ -343,6 +343,50 @@ public void addDataPartitionedHive() {
         sql("SELECT id, name, dept, subdept FROM %s ORDER BY id", tableName));
   }
 
+  @Test
+  public void addDataPartitionedHiveInParallel() {
+    createPartitionedHiveTable();
+
+    String createIceberg =
+            "CREATE TABLE %s (id Integer, name String, dept String, subdept String) USING iceberg PARTITIONED BY (id)";
+
+    sql(createIceberg, tableName);
+
+    Object result = scalarSql("CALL %s.system.add_files(" +
+                    "table => '%s', " +
+                    "source_table => '%s', " +
+                    "list_partition_parallelism => 3)",
```

Comment:
would be nice if we had a way to actually test that this was working properly and not just not throwing an error ... Can't think of an easy way at the moment so this is fine